pkg/kn/commands/root.go

```diff
@@ -129,3 +138,21 @@ func GetConfig() (serving.ServingV1alpha1Interface, error) {
 	}
 	return client, nil
 }
+
+func setNamespace() {
+	if namespace != "" {
+		return
+	}
+	namespace = "default"
+	config := clientConfig(kubeCfgFile)
+	if ns, _, err := config.Namespace(); err == nil {
+		namespace = ns
+	}
+}
+
+func clientConfig(path string) clientcmd.ClientConfig {
```

Comment:
here is a better version i cooked up for [kapp](https://github.com/k14s/kapp): https://github.com/k14s/kapp/blob/master/pkg/kapp/cmd/core/config_factory.go#L71-L82. it properly deals with multi-value KUBECONFIG values and picks up default env vars based on cmdclient config. (need to update knctl to do so as well...)